### PR TITLE
chore(deps): bump happy-dom 15→20, vitest 2→4, @vitest/coverage-v8 2→4

### PR DESCRIPTION
**Pull request by clintongormley**
## Summary

Supersedes dependabot PRs #478, #479, #480 with a single self-contained bump that keeps the 90% coverage gate green.

- `happy-dom`: ^15.0.0 → ^20.9.0
- `vitest`: ^2.1.0 → ^4.1.6
- `@vitest/coverage-v8`: ^2.1.0 → ^4.1.6
- Transitive: `vite` 5 → 8, `esbuild` updated

## Why the dependabot PRs failed CI

Vitest 4's `@vitest/coverage-v8` uses AST-aware remapping by default, which counts the same code as having more (and finer-grained) branches than Vitest 2's native v8 mapping. On this codebase the same source dropped from **90.67% → 84.82%** branch coverage on the bump alone — not a real regression, just a more accurate count.

## What this PR does

1. **`/* v8 ignore */` on module-load registration blocks** in all three card files. These are defensive duplicate-registration guards (`if (!customElements.get(...))`) whose "already defined" branch can't be exercised in a single-process test run. ~50 untestable branches removed from the denominator.

2. **Edge-case test additions** for branches v4 now flags as genuine gaps — `lang || "en"` defaults, optional ternary else paths, null-guard branches in gesture/PIN handlers, refresh fallback timers. Three new `*-edges.test.js` integration files plus unit-test extensions for `card-utils` and `activity-log` helpers.

## Coverage after the bump

| File | Branches (was → now) |
|---|---|
| `verisure-owa-card-utils.js` | 93.33% → **100%** |
| `verisure-owa-camera-card.js` | 82.50% → **92.59%** |
| `verisure-owa-activity-log-card.js` | 84.87% → **90.43%** |
| `verisure-owa-alarm-card.js` | 85.11% → **89.07%** |
| **Global** | **84.82% → 90.12%** |

All four metrics ≥ 90% gate: 97.79% stmts / 90.12% branches / 96.65% funcs / 99.24% lines.

## Test plan

- [x] `npm install` clean (one pre-existing eslint-plugin-vitest peer-dep warning, unrelated)
- [x] `npm test` — 344 passed (286 existing + 58 new edge-case tests)
- [x] `npm run test:coverage` — gate passes
- [x] `npm run lint` — clean (2 pre-existing warnings in alarm-card.js, unrelated)
- [x] `ruff check` / `ruff format` — clean
- [x] `pyright` — 0 errors
- [x] `pylint` — 10.00/10
- [x] `pytest` — 1573 passed

## Follow-up

Once this merges, the three superseded dependabot PRs (#478 happy-dom, #479 vite+vitest, #480 esbuild+vitest) should be closed — this PR's bumps cover all of them.

### CODE REVIEW

File: tests-js/integration/activity-log-card-edges.test.js [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  });
+});
+
+describe("activity-log-card defaults / fallbacks", () => {
+  it("getCardSize defaults to 11 cells when no limit configured", () => {
+    const card = mountActivityCard({
+      hass: makeHass({ states: { [ENTITY]: makeActivityLogEntity({ events: [] }) } }),
+    });
+    expect(card.getCardSize()).toBe(8);
+  });
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply tests-js/integration/activity-log-card-edges.test.js]: Fixed in 8073516. The actual behaviour is  — default limit 10 yields min(8, 11) = 8, so the test was correct but the description was misleading. Renamed to clarify the cap.

File: tests-js/integration/activity-log-card-edges.test.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+      hass: makeHass({
+        states: { [ENTITY]: makeActivityLogEntity({ events: [] }) },
+      }),
+    });
+    card.remove();
+    // No assertion needed — the disconnectedCallback path is what we want
+    // covered; if it doesn't clear the timer, leaked timers would surface
+    // as cross-test interference.
+    vi.advanceTimersByTime(60_000);
+    expect(true).toBe(true);
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply tests-js/integration/activity-log-card-edges.test.js]: Fixed in 8073516. Now asserts `_tickTimer` is null after `remove()` and spies on `_render` to prove the interval callback no longer fires.

**Comment by clintongormley** [comment]
@guerrerotook - ready for your approval, thanks

**Review by guerrerotook — APPROVED** [review]
